apps/src/p5lab/P5Lab.js

```diff
@@ -519,6 +505,66 @@ P5Lab.prototype.init = function(config) {
   return loader;
 };
 
+/**
+ * Load any necessary missing animations. For now, this is mainly for
+ * the set background to block, which needs there to be backgrounds in the
+ * animation list at the start in order to look not broken,
+ * @param {Boolean} useConfig
+ * @param {Object} initialAnimationList
+ */
+P5Lab.prototype.loadMissingAnimationDefaultsIfNecessary = function(
+  useConfig,
+  initialAnimationList
+) {
+  if (this.isSpritelab) {
+    let configDictionary = {};
+    initialAnimationList.orderedKeys.forEach(key => {
+      const name = initialAnimationList.propsByKey[key].name;
+      configDictionary[name] = key;
+    });
+    if (useConfig && this.startAnimations) {
+      // We need to make sure we include all of the default animations (in case they have changed since this project was created).
+      this.startAnimations.orderedKeys.forEach(key => {
+        const name = this.startAnimations.propsByKey[key].name;
+        if (!configDictionary[name]) {
+          initialAnimationList.orderedKeys.push(key);
+          initialAnimationList.propsByKey[
+            key
+          ] = this.startAnimations.propsByKey[key];
+        }
+      });
+    }
+    // Check if initialAnimationList has backgrounds. If it doesn't add it from defaultSprites.json
+    // This is primarily to handle pre existing levels that don't have animations in their list yet
+    let hasBackgrounds = false;
+    initialAnimationList.orderedKeys.forEach(key => {
+      if (
+        initialAnimationList.propsByKey[key].categories &&
+        initialAnimationList.propsByKey[key].categories.includes('backgrounds')
+      ) {
+        hasBackgrounds = true;
+      }
+    });
+    defaultSprites.orderedKeys.forEach(key => {
+      if (
+        defaultSprites.propsByKey[key].categories.includes('backgrounds') &&
+        configDictionary[defaultSprites.propsByKey[key].name]
+      ) {
+        hasBackgrounds = true;
+      }
+    });
```

Comment:
i think we can simplify both `forEach` loops here to make the intent clearer by using `filter` instead. the first loop could look something like:

```javascript
initialAnimationList.orderedKeys.filter(key => {
  const {categories} = initialAnimationList.propsByKey[key];
  return categories && categories.includes('backgrounds');
});
```

then we check whether either resulting array has any items, which means we have backgrounds:

```javascript
const hasBackgrounds = firstFilter.length > 0 || secondFilter.length > 0;
```